### PR TITLE
added macro documentation

### PR DESCRIPTION
**Pull request by adhakal321**
## Pull Request Purpose

- [ ] bug fix PR with no breaking changes — please ensure the base branch is `master`
- [x] new functionality — please ensure the base branch is the latest `main` branch
- [ ] a breaking change — please ensure the base branch is the latest `main` branch

## Checklist

- [ ] I have verified that these changes work locally via my sandbox
- [ ] Added tests & descriptions to my macros (and models if applicable)
- [ ] Pushed code into test
- [x] Updated the README.md (if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: docs/macros_schema.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{% docs ref %}
+    This macro is the default dbt macro  for the model. ref is used in the sql file to refer to the model name.
```
Comment (jonhopper-dataengineers) [inline docs/macros_schema.md:2]: This macro overrides the default 'ref' dbt macro when in your local dbt project. This is used in your models to reference other models

File: docs/macros_schema.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    example:
+    
+        ref('stg_payments') -- where stg_payments is the ref model name
+        or
+        ref('stg_payments', true) -- where raw_customers is the source model and true is to include the database name
+
+{% enddocs %}
+
+{% docs src %}
+    This macro is the default dbt macro  for the model. source is used in the sql file to refer to the source model name.
```
Comment (jonhopper-dataengineers) [inline docs/macros_schema.md:13]: This macro overrides the default 'source' dbt macro when in your local dbt project. This is used in your models to reference source models

File: docs/macros_unit_tests.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{% docs unit_test_description %}
+    describes the atomic unit test that need to be run:
```
Comment (jonhopper-dataengineers) [inline docs/macros_unit_tests.md:2]: This macro describes an atomic unit test that is executed when running 'dbt test'

File: docs/macros_unit_tests.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            - name: customer_id
+                description: This is a unique identifier for a customer
+                tests:
+                - unique
+                - not_null
+
+{% enddocs %}
+
+{% docs input_mapping_description %}
+    descibes macro should be replacing the source or references in the test
```
Comment (jonhopper-dataengineers) [inline docs/macros_unit_tests.md:26]: specifies which 'ref' or 'source' macros should be replaced with the relevant test macro

File: docs/macros_unit_tests.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    example:
+
+     ref('dim_customers'): "customers macro name"
+     ref('fct_orders'): "prders macro name"
+
+{% enddocs %}
+
+{% docs expected_output %}
+    descibes macro should be replacing the source or references in the test
```
Comment (jonhopper-dataengineers) [inline docs/macros_unit_tests.md:36]: specifies the expected output of the model. This needs to be a table and not a macro.

File: macros/atomic-unit-tests/atomic-unit-tests.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        description: name of the unit test. This should be the same as the test case folder under the 'unit tests' folder. Its format shoul be 'tc__<<model name>>_<<test name>>'  example - tc__fct_orders_has_all_payment_methods
+      - name: description
+        type: string
+        description: readable description of test case
+      - name: compare_columns
+        type: List
+        description: List of  It allows to compare specified columns otherwise compare all columns.
+  
+  - name: get_unit_test_sql
+    description: Generates the SQL for the unit test
```
Comment (jonhopper-dataengineers) [inline macros/atomic-unit-tests/atomic-unit-tests.yml:29]: This is for internal use to the test_unit_test macro. The macro generates the SQL for the unit test

File: macros/grants/grants.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
+version: 2
+
+macros:
+  - name: grant_database_ownership_access
+    description: This macro grants ownership privilege to the specific roles within the database
```
Comment (jonhopper-dataengineers) [inline macros/grants/grants.yml:5]: specific roles? or specified rolename

File: macros/grants/grants.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    docs:
+      show: true
+    arguments:
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+
+  - name: grant_database_usage_access
+    description: This macro grants database usages privilege to the specific roles within the database. To get access on the database, the role needs to have usage privilege in the database.
```
Comment (jonhopper-dataengineers) [inline macros/grants/grants.yml:14]: specific roles? or specified rolename

File: macros/grants/grants.yml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        description: Name of the schema
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+      - name: include_future_grants
+        type: bool
+        description: Grants future privilege to the objects FUTURE VIEWS, FUTURE MATERIALIZED VIEWS, FUTURE TABLES, and FUTURE EXTERNAL TABLES. Default is false
+
+  - name: grant_schema_read_access
+    description: This macro grants read-only access to the specified role inside the schema.
```
Comment (jonhopper-dataengineers) [inline macros/grants/grants.yml:38]: This macro grants read-only access to the specified rolename inside the schema.

File: macros/grants/grants.yml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        description: Name of the schema
+      - name: rolename
+        type: string
+        description: Name of the role eg- ANALYST
+      - name: include_future_grants
+        type: bool
+        description: Grants future privilege to the objects FUTURE VIEWS, FUTURE MATERIALIZED VIEWS, FUTURE TABLES, and FUTURE EXTERNAL TABLES. Default is false
+  
+  - name: grant_schema_write_access
+    description: This macro grants write access to the specific role inside the schema.
```
Comment (jonhopper-dataengineers) [inline macros/grants/grants.yml:53]: This macro grants write access to the specified rolename inside the schema.

File: macros/grants/grants.yml (hunk 8 above)
Comment (jonhopper-dataengineers) [reply macros/grants/grants.yml:14]: This macro grants usage privilages to the specified rolename inside the database. To get access on the database, the rolename needs to have usage privilege in the database.

File: macros/grants/grants.yml (hunk 7 above)
Comment (jonhopper-dataengineers) [reply macros/grants/grants.yml:5]: This macro grants ownership privileges to the specified rolename on the database.

File: macros/modelling/modelling.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        description:  the date is passed as an argument, the format is YYYYMMDD
+      - name: timeKey
+        type: string
+        description: time is passed as an argument, the format is HHMI
+      - name: dt_format
+        type: string
+        description: converts the date-time into specified DateTime format otherwise 'YYYYMMDDHHMI' as a date format
+
+  - name: datetime_to_date_dim
+    description: This macro converts the specified col data into a Date with the format 'YYYYMMDD'
```
Comment (jonhopper-dataengineers) [inline macros/modelling/modelling.yml:29]: change data to fields

File: macros/modelling/modelling.yml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    docs:
+      show: true
+    arguments:
+      - name: col
+        type: string
+        description: The Name of the column
+
+  - name: datetime_to_time_dim
+    description: This macro converts the specified col into a Time with the format HHMI
```
Comment (jonhopper-dataengineers) [inline macros/modelling/modelling.yml:38]: This macro converts the specified col field into a Time with the format 'HHMI'

File: macros/modelling/modelling.yml [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    docs:
+      show: true
+    arguments:
+      - name: col
+        type: string
+        description: The Name of the column
+
+  - name: time_key
+    description: This macro converts the specified TimeKey into the time key(HHMI format).
```
Comment (jonhopper-dataengineers) [inline macros/modelling/modelling.yml:47]: This macro converts the specified col field into a Time with the format 'HHMI'

**Review by jonhopper-dataengineers — CHANGES_REQUESTED** [review]

File: docs/macros_schema.md (hunk 1 above)
Comment (adhakal321) [reply docs/macros_schema.md]: donw

File: docs/macros_schema.md (hunk 2 above)
Comment (adhakal321) [reply docs/macros_schema.md]: done

File: docs/macros_unit_tests.md (hunk 3 above)
Comment (adhakal321) [reply docs/macros_unit_tests.md]: done

File: docs/macros_unit_tests.md (hunk 4 above)
Comment (adhakal321) [reply docs/macros_unit_tests.md]: done

File: docs/macros_unit_tests.md (hunk 5 above)
Comment (adhakal321) [reply docs/macros_unit_tests.md]: done

File: macros/atomic-unit-tests/atomic-unit-tests.yml (hunk 6 above)
Comment (adhakal321) [reply macros/atomic-unit-tests/atomic-unit-tests.yml]: done

File: macros/grants/grants.yml (hunk 7 above)
Comment (adhakal321) [reply macros/grants/grants.yml]: done

File: macros/grants/grants.yml (hunk 8 above)
Comment (adhakal321) [reply macros/grants/grants.yml]: done

File: macros/grants/grants.yml (hunk 9 above)
Comment (adhakal321) [reply macros/grants/grants.yml]: done

File: macros/grants/grants.yml (hunk 10 above)
Comment (adhakal321) [reply macros/grants/grants.yml]: done

File: macros/modelling/modelling.yml (hunk 11 above)
Comment (adhakal321) [reply macros/modelling/modelling.yml]: done

File: macros/modelling/modelling.yml (hunk 12 above)
Comment (adhakal321) [reply macros/modelling/modelling.yml]: done

File: macros/modelling/modelling.yml (hunk 13 above)
Comment (adhakal321) [reply macros/modelling/modelling.yml]: done

**Review by jonhopper-dataengineers — APPROVED** [review]

**Pull request merged**